### ISSUE TITLE
Style with multiple setters in XAML won't compile

### ISSUE THREAD

**Issue by nicolas-garcia** [xamarin/Xamarin.Forms#1825]
### Description
Hi!  
I am new to Xamarin.Form, I come from WPF world. Currently, I struggle with this issue, and I am not sure its a real one. Am I missing something huge? If so, I am so sorry for opening this issue.

### Steps to Reproduce

1. Create a new Xamarin.Form project, .Net Standard 2.0
2. In `App.xaml`, add a new style with two setters:
```xaml
<?xml version="1.0" encoding="utf-8" ?>
<Application xmlns="http://xamarin.com/schemas/2014/forms"
             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
             x:Class="TestBugXamarin.App">
    <Application.Resources>
        <ResourceDictionary>
            <Style x:Key="MyLabelStyle" TargetType="{x:Type Button}">
                <Style.Setters>
                    <Setter Property="TextColor" Value="White" />
                    <Setter Property="FontSize" Value="16" /> <!-- Comment this line and the XAML compile. -->
                </Style.Setters>
            </Style>
        </ResourceDictionary>
	</Application.Resources>
</Application>
```
3. In `App.xaml.cs`, add the xaml compilation annotation:
```csharp
using Xamarin.Forms.Xaml;

[assembly: XamlCompilation(XamlCompilationOptions.Compile)]
namespace TestBugXamarin
{
   // default code here
}
```

### Expected Behavior

The project should compile.

### Actual Behavior

Compilation error: `App.xaml : error : Object reference not set to an instance of an object.`  
If I remove the second setter in style, it works.

### Basic Information

- Version with issue: 2.5.0.122203
- Last known good version: None
- IDE: Visual Studio Professional 2017 v15.5.6
- Nuget Packages: Xamarin.Form v2.5.0.122203

### Reproduction Link

[TestBugXamarin.zip](https://github.com/xamarin/Xamarin.Forms/files/1702556/TestBugXamarin.zip)

**Comment by StephaneDelcroix** [comment]
I can confirm the bug on the version you're using (-sr3). It's no longer there on `2.6.0.162115-nightly`. I'm trying to figure if the fix for this is present in the upcoming -sr4 or not...

**Comment by StephaneDelcroix** [comment]
As a workaround, you can skip the `<Style.Setters>` part. it's implicit, and make your code works

```xml
<Style x:Key="MyLabelStyle" TargetType="{x:Type Button}">
                    <Setter Property="TextColor" Value="White" />
                    <Setter Property="FontSize" Value="16" /> <!-- Comment this line and the XAML compile. -->
</Style>
```

**Comment by StephaneDelcroix** [comment]
this was fixed in #1228

**Comment by nicolas-garcia** [comment]
Thank you for the workaround!
